sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/Analytics.java

```diff
@@ -749,7 +750,7 @@ private void queuePage(String name, Map<String, String> properties) {
      * @param flags              optional flags.
      */
     private synchronized void trackEventAsync(final String name, final List<TypedProperty> properties, final AnalyticsTransmissionTarget transmissionTarget, final int flags) {
-        final String userId = AppCenter.getInstance().getUserId();
+        final String userId = UserIdContext.getInstance().getUserId();
```

Comment:
Not related to this PR but userId assignment could be done in Channel since all logs have userId property.

Reply:
Its not the spec though.